pkg/operator/ceph/pool/controller.go

```diff
@@ -497,3 +502,45 @@ func (r *ReconcileCephBlockPool) cancelMirrorMonitoring(cephBlockPool *cephv1.Ce
 		delete(r.blockPoolContexts, channelKey)
 	}
 }
+
+func (r *ReconcileCephBlockPool) disableMirroring(pool *cephv1.CephBlockPool) error {
+	// Remove storage cluster peers
+	mirrorInfo, err := cephclient.GetPoolMirroringInfo(r.context, r.clusterInfo, pool.Name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get mirroring info for the pool %q", pool.Name)
+	}
+	for _, peer := range mirrorInfo.Peers {
+		if peer.UUID != "" {
+			err := cephclient.RemoveClusterPeer(r.context, r.clusterInfo, pool.Name, peer.UUID)
+			if err != nil {
+				return errors.Wrapf(err, "failed to remove cluster peer with UUID %q for the pool %q", peer.UUID, pool.Name)
+			}
+			logger.Infof("successfully removed peer site %q for the pool %q", peer.UUID, pool.Name)
+		}
+	}
+
+	if mirrorInfo.Mode == "image" {
+		mirroringStatus, err := cephclient.GetPoolMirroringStatus(r.context, r.clusterInfo, pool.Name)
+		if err != nil {
+			return errors.Wrapf(err, "failed to get images for pool %q", pool.Name)
+		}
+
+		for _, image := range *mirroringStatus.Images {
+			// Disable mirroring on pool image
+			err = cephclient.DisablePoolImageMirroring(r.context, r.clusterInfo, pool.Name, image.Name)
+			if err != nil {
+				return errors.Wrapf(err, "failed to disable mirroring for image %q in pool %q", image.Name, pool.Name)
+			}
+			logger.Infof("successfully disabled mirroring on image %q for the pool %q", image.Name, pool.Name)
+		}
```

Comment:
if the user enabled the mirror for each image, why not let the user disable the mirroring or let rook disable only when certain flags are set on the pool CR?

Reply:
After discussion, not removing the mirrored images anymore. 
When support for cleanup annotation is implemented, then images can be deleted if annotation is present.